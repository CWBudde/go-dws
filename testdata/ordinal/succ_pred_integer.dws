// Test Succ() and Pred() with integers
// Tasks 9.28 and 9.29: Successor and predecessor functions

var x: Integer := 10;
PrintLn('Initial value: ', x);

var next: Integer := Succ(x);
PrintLn('Succ(x): ', next);

var prev: Integer := Pred(x);
PrintLn('Pred(x): ', prev);

// Chain Succ calls
PrintLn('Succ(Succ(x)): ', Succ(Succ(x)));

// Chain Pred calls
PrintLn('Pred(Pred(x)): ', Pred(Pred(x)));

// Test with zero
PrintLn('Succ(0): ', Succ(0));
PrintLn('Pred(0): ', Pred(0));

// Test with negative
var neg: Integer := -5;
PrintLn('Succ(-5): ', Succ(neg));
PrintLn('Pred(-5): ', Pred(neg));

// Combined with Inc/Dec
Inc(x);
PrintLn('After Inc(x): ', x);
PrintLn('Succ(x): ', Succ(x));

Dec(x);
PrintLn('After Dec(x): ', x);
PrintLn('Pred(x): ', Pred(x));
