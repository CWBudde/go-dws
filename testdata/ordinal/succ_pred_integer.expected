Initial value:  10
Succ(x):  11
Pred(x):  9
Succ(Succ(x)):  12
Pred(Pred(x)):  8
Succ(0):  1
Pred(0):  -1
Succ(-5):  -4
Pred(-5):  -6
After Inc(x):  11
Succ(x):  12
After Dec(x):  10
Pred(x):  9
